stately-concurrent-collections/src/commonMain/kotlin/co/touchlab/stately/collections/ConcurrentMutableList.kt

```diff
@@ -0,0 +1,65 @@
+package co.touchlab.stately.collections
+
+import co.touchlab.stately.concurrency.Synchronizable
+import co.touchlab.stately.concurrency.synchronize
+
+class ConcurrentMutableList<E> internal constructor(rootArg: Synchronizable?, private val del: MutableList<E>) :
+    ConcurrentMutableCollection<E>(rootArg, del), MutableList<E> {
+    constructor() : this(null, mutableListOf())
```

Comment:
Why does this one have an extra constructor instead of default parameters (comparing to `ConcurrentMutableCollection`)?

Reply:
All of the more concrete ones have the second constructor. the main one is internal. The `ConcurrentMutableCollection` constructor should probably be internal as you shouldn't really ever create one of those directly.

Reply:
That makes sense.

Reply:
Is there a usecase in which you'd want to wrap an existing instance of a List (or another collection type) in the Concurrent type? Like doing some work in parallel, but then passing the collection back to be used by single-threaded operations only?

Reply:
Don't know. For now I think I'd leave it as is. You can access the mutable collection in a block if needed. If somebody has a real need we can alter it.